### DISCUSSION TITLE
NordicTrack GX 4.4 Pro with QZ on Samsung S7 (Android 8)

### DISCUSSION THREAD

**Discussion by aloehr70** [cagnulein/qdomyos-zwift#2836] [Q&A]
Hi, I am running QZ on an old Samsung S7 (Android 8) with a NordicTrack GX 4.4 Pro. With the default settings I get power and cadence from the bike but I am not able to set the resistance from QZ. Maybe someone can tell me which settings I have to alter to be able to set the resistance from QZ? Thanks in advance! Regards, Alex

**Comment by cagnulein** [comment]
hi @aloehr70 you have to follow this with the ifit app https://github.com/cagnulein/qdomyos-zwift/blob/master/docs/20_supported_devices_and_applications.md#android-device to collect a log from it and so i can implement as well on qz
let me know!

**Reply by aloehr70** [reply]
Hi,

here are two files. One with a log when I set the resistance on the bike and one file when I set the resistance in the iFit app on an iPad. Thank you very much for your efforts.

[bugreport-2024-11-27-10-10-30_SettingResistanceInIFitAppOnIPad.zip](https://github.com/user-attachments/files/17931977/bugreport-2024-11-27-10-10-30_SettingResistanceInIFitAppOnIPad.zip)
[bugreport-2024-11-27-10-00-11_SettingResistanceOnBike.zip](https://github.com/user-attachments/files/17931978/bugreport-2024-11-27-10-00-11_SettingResistanceOnBike.zip)

**Reply by cagnulein** [reply]
ok thanks, i will check it asap, i'm currently overwhelmed by requests, it could take some days/weeks

**Reply by cagnulein** [reply]
@aloehr70 i checked quickly the log but the hci snoof file is missing. did you enable it on your phone? it's required for this unfortunately

**Reply by aloehr70** [reply]
Excuse me, my fault. I didn't restart the phone. This report now contains a btsnoop_hci.log. This time I first set the resistance on the bike starting from 1 up to 25 (max) and decreases it step by step back to 1 and then I used the iFit App to change the resistance. Once more: Thank you very much!

[bugreport-2024-11-27-11-01-47.zip](https://github.com/user-attachments/files/17932583/bugreport-2024-11-27-11-01-47.zip)

**Reply by cagnulein** [reply]
thanks @aloehr70 now i see the file but i don't see any ifit communication there. Just to be sure are you running ifit on this android device? it must be done in this way
Let me know!

**Reply by aloehr70** [reply]
Sorry. I now installed the ifit app on my samsung s7 and logged the bluetooth stuff on it. I hope it now contains the proper data. I changed the resistance on my bike from 1, 2, 3, 4, 5, 4, 3, 2,1 (I think in this case the bike sends the resistance value to the ifit app) and I changed the resistance in the ifit app from 1, 2, 3, 4, 5, 4, 3, 2, 1 (I think in this case the ifit app sends the resistance value to the bike). And agian: Thank you very much!

[bugreport-2024-11-27-14-24-30.zip](https://github.com/user-attachments/files/17935082/bugreport-2024-11-27-14-24-30.zip)

**Reply by cagnulein** [reply]
yes seems perfect now!

**Comment by cagnulein** [comment]
ok @aloehr70 i did a very first raw version

https://github.com/cagnulein/qdomyos-zwift/actions/runs/12066356966

Google is very slow in these days, reviewing the app on the store, even the beta ones. For this reason, in order to check right now if the patch that I did is right, you can use the android build in the link (check at the bottom of the page in about 1 hour). In order to test it you need first to remove the current QZ version on your phone, install the one that you downloaded.
This version is a 15 minutes trial (you can restart it as many times as you want) and I can unlock it if you want, send me an email to roberto.viola83@gmail.com mentioning this ticket and that you are on android.
Let me know if you have any questions

select the new nordictrack bike gx 4.4 pro in the combo box in the nordictrack bike settings in qz and restart it
let me know!

**Comment by aloehr70** [comment]
I was able to test your work but I believe it doesn't work as expected. I
used the tiles resistance +/- buttons on the screen but nothing happens.

I uninstalled my version from the Google play store, installed the
android-debug.apk and set Nordic track GX 4.5 in bike options. No luck. The
cadence and other things are received but the resistance did not change at
all. Even when I changed the resistance on the bike. Which button in your
app do I have to use to change the bike resistance?
---
Alexander Löhr
Engelbrechtsweg 2a
D-33161 Hövelhof
+49 160 960 11 914

Roberto Viola ***@***.***> schrieb am Do., 28. Nov. 2024,
10:46:

**Comment by cagnulein** [comment]
You have to use the resistance plus and minus buttons in qz
If it doesn't work collect a ifit log where

You are changing from the ifit app the resistance to 1 to max and again to
1 step by step wirh a 5 seconds delay between each change

Let me know!

Roberto Viola
Software engineer and open source enthusiast
http://robertoviola.cloud



Il giorno gio 28 nov 2024 alle 13:07 aloehr70 ***@***.***> ha
scritto:
